cudaextender/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+#
+# Copyright 2019-2020 NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+project(cudaextender)
+
+# Process data subdirectory first
+add_subdirectory(data)
+
+GitVersion()
+
+configure_file(${CMAKE_SOURCE_DIR}/common/base/src/version.cpp.in
+        ${CMAKE_CURRENT_BINARY_DIR}/version.cpp)
+
+# Optimization flags interfere with nvcc and cause compilation issues with reworked
+# file structure.
+if (NOT DEFINED gw_cuda_before_10_0)
+    message(FATAL_ERROR "${PROJECT_NAME} : variable gw_cuda_before_10_0 is not defined yet. Please make sure CUDA.cmake is loaded first.")
+elseif (gw_cuda_before_10_0)
+    message(STATUS "${PROJECT_NAME} : Remove -O optimization when building for CUDA < 10 as it causes compilation issues.")
```

Comment:
This is only relevant for `cudapoa`. I recall the error message being something along the lines of picking up some intel sse headers when the optimization flag was enabled. It only happened after I did some refactoring, and only showed up with cuda 9 + -O flags. Put in the WAR since we were going to drop cuda 9 support anyway

Reply:
Dropped it from cudaextender.